procedures/unit-testing-basics.ipf

```diff
@@ -1238,8 +1245,420 @@ static Function ExecuteHooks(hookType, hooks, juProps, name, procWin, [param])
 	endswitch
 End
 
+/// @brief Background monitor of the Unit Testing Framework
+Function $"UTFBackgroundMonitor"(s)
+	STRUCT WMBackgroundStruct &s
+
+	variable i, numTasks, result, stopState
+	string task
+
+	DFREF df = GetPackageFolder()
+	SVAR/Z tList = df:BCKG_TaskList
+	SVAR/Z rFunc = df:BCKG_ReentryFunc
+	NVAR/Z timeout = df:BCKG_EndTime
+	NVAR/Z mode = df:BCKG_Mode
+
+	if(!SVAR_Exists(tList) || !SVAR_Exists(rFunc) || !NVAR_Exists(mode) || !NVAR_Exists(timeout))
+		print "Fatal: UTFBackgroundMonitor can not find monitoring data in package DF, aborting monitoring."
+		return 2
+	endif
+
```

Comment:
@MichaelHuth You can do the timeout check here and call `RunTest`. Do we want a `print` for the timeout case as well? I'm not sure.